src/duqtools/ids/_mapping.py

```diff
@@ -356,6 +354,18 @@ def to_dataframe(self,
 
         return df
 
+    def _fill_array_from_partial_path(self, partial_path):
+        arr = []
+        for index in range(len(self[partial_path[0]])):
+            path = partial_path[0] + '/' + str(index) + '/' + partial_path[1]
+            if len(partial_path) > 2:
+                sub_partial_path = [path] + partial_path[2:]
+                sub_arr = self._fill_array_from_partial_path(sub_partial_path)
+                arr.append(sub_arr)
+            else:
+                arr.append(self[path])
+        return arr
+
```

Comment:
I found this function a bit tricky to read, so I had some fun trying to refactor while trying to understand how it works. Hope you don't mind :-)

```suggestion
    def _fill_array_from_parts(self, *parts: str):
        arr = []
        root, sub, *remaining = parts
        nodes = self[root]
        
        for index in range(len(nodes)):
            path = f'{root}/{index}/{sub}'
            
            if remaining:
                sub_arr = self._fill_array_from_parts(path, *remaining)
            else:
                sub_arr = self[path]
            
            arr.append(sub_arr)

        return arr
```

Reply:
So much more pythonic, It looks much better,  I guess I am really more used in doing thing the "old" way.